addons/portal/views/portal_templates.xml

```diff
@@ -5,6 +5,9 @@
             <attribute name="t-attf-class" add="#{request.env['res.lang']._get_data(code=request.env.lang).direction == 'rtl' and 'o_rtl' or ''}" separator=" "/>
             <attribute name="t-attf-class" add="#{'o_portal' if is_portal else ''}" separator=" "/>
         </xpath>
+        <xpath expr="//div[@id='wrapwrap']" position="before">
```

Comment:
It looks good to me. Just one last thing (sorry): actually it would be better inside the `#wrapwrap` (because the scrollbar is on the wrapwrap, not on the body), so that it doesn't stay on the window when you scroll the page.